Automattic/newspack-blocks
# Refactor the Donate block in preparation for more layout options

### All Submissions:






### Changes proposed in this Pull Request:



A rather large refactor which lays the groundwork for additional layout options for the Donate block. 

### How to test the changes in this Pull Request:

1. Insert the Donate block on a page, observe the donations parameters are as provided in the Reader Revenue wizard 
2. Publish the page, make a donation using all three Reader Revenue platforms 
3. Override the block settings, test donation flow again

### Other information:
